conductor-client/src/main/kotlin/com/openlattice/data/jobs/DataDeletionJob.kt

```diff
@@ -106,6 +92,13 @@ class DataDeletionJob(
             return
         }
 
+        // batch delete neighbors first. Only delete from state.entitySetId if there are no more neighbors to delete
+        val neighborEntityDataKeys = getBatchOfNeighborEntityDataKeys(entityDataKeys.map { it.entityKeyId }.toSet())
+        if (neighborEntityDataKeys.isNotEmpty()) {
+            deleteNeighborEntitiesAndEdges(neighborEntityDataKeys)
+            return
+        }
```

Comment:
The result of getBatchOfEntityDataKeys() is deleted after neighbors are deleted.

Reply:
Logic looks ok to me, but is the `return` necessary?

Reply:
Looks good to me.

- I believe `return` is necessary so that when `processNextBatch()` is called, only a single batch of neighbors is deleted per function call as long as neighbors remain. While neighbors still remain we do not want to delete entities. Once all neighbors have been deleted, return is skipped as we now are okay to begin deleting all entities.
- And 104-110 are used to exit the function if there is no work to be done.